### PR TITLE
ExteriorCovering and SearchRanges

### PR DESCRIPTION
**Pull request by davidmcleish**
Add `ExteriorCovering` that approximates a simplefeatures `Geometry` as a set of `QuadKey`s.

Also added `SearchRanges` which, given a set of `QuadKey`s, returns the ranges of indices to search for that contain those tiles and any of their descendants.

### CODE REVIEW

File: quadtree/quadkey.go [hunk 1]
```diff
@@ -8,18 +8,21 @@ import (
...
 // Parent get parents quadkey for passed quadkey
 func (q QuadKey) Parent() (QuadKey, error) {
 	zoomLevel := q.Zoom()
-	parentZoomLevel := zoomLevel - 1
-
-	if parentZoomLevel <= 0 {
+	if zoomLevel <= 0 {
 		return 0, errors.New("no parent")
 	}
+	parentZoomLevel := zoomLevel - 1
```
Comment (kpfaulkner) [inline quadtree/quadkey.go:25]: ummm that will allow ParentZoomLevel to be -1.

Comment (davidmcleish) [reply quadtree/quadkey.go:25]: less than or equal to

Comment (davidmcleish) [reply quadtree/quadkey.go:25]: It might be clearer to keep the original and change it to
```go
if parentZoomLevel < 0 {
```

**Review by kpfaulkner — APPROVED** [review]
LGTM

File: quadtree/quadkey.go (hunk 1 above)
Comment (davidmcleish) [reply quadtree/quadkey.go:25]: Ah, I knew there was a reason I did it this way. `byte` is unsigned, so
```go
if parentZoomLevel < 0 {
```
is trivially false.

**Pull request merged**